### PR TITLE
HSEARCH-3485 + HSEARCH-3484 JDK12 compatibility

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-3485
https://hibernate.atlassian.net/browse/HSEARCH-3484

Full build including a build with JDK12: http://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-1885/2/pipeline/66

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1771,6 +1771,38 @@
...
+                </surefire.jvm.args.java-version.strict>
+                <arquillian.wildfly.jvm.args.java-version>
+                    -Djdk.attach.allowAttachSelf=true
+                </arquillian.wildfly.jvm.args.java-version>
+            </properties>
+            <modules>
+                <!-- Nothing at the moment -->
+            </modules>
+        </profile>
+
```
Comment (gsmet) [inline pom.xml]: I think I would just have made the version of the above profile an open bound. Not sure it's worth having the exact same profile for N+ JDK.

File: Jenkinsfile [hunk 2]
```diff
@@ -394,7 +394,7 @@ stage('Default build') {
 					"} \
 					-Pdist -Pcoverage -Pjqassistant \
 					${enableDefaultEnvIT ? '' : '-DskipITs'} \
-					${enableDefaultEnvLegacyIT ? '-Dsurefire.legacy.skip=false -Dfailsafe.legacy.skip=false' : ''} \
+					${enableDefaultEnvLegacyIT ? '-Dlegacy.skip=false' : ''} \
```
Comment (gsmet) [inline Jenkinsfile]: Ah nice, I suggested that a while ago, it's definitely better.

File: pom.xml (hunk 1 above)
Comment (yrodiere) [reply pom.xml]: I didn't know we could do that. You're right, it would be better; will do.

**Comment by yrodiere** [comment]
Updated according to your comment and launched a full build. Waiting for CI.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**